bilby/gw/likelihood.py

```diff
@@ -1452,17 +1452,20 @@ def __init__(self, interferometers, waveform_generator,
         self.per_detector_fiducial_waveforms = {}
         self.bin_freqs = dict()
         self.bin_inds = dict()
-        self.initial_parameter_keys_sorted = sorted(self.initial_parameters)
-        self.maximum_likelihood_parameters = initial_parameters
-        self.set_fiducial_waveforms(self.initial_parameters)
+        self.set_fiducial_waveforms(self.fiducial_parameters)
         logger.info("Initial fiducial waveforms set up")
         self.setup_bins()
         self.compute_summary_data()
         logger.info("Summary Data Obtained")
+        self.parameters_to_be_updated = sorted({"chirp_mass", "mass_ratio", "a_1", "a_2", "tilt_1",
+                                                "tilt_2", "phi_12", "phi_jl", "lambda_1", "lambda_2"})
+
+        if update_fiducial_parameters:
+            self.fiducial_parameters = self.find_maximum_likelihood_parameters(self.parameter_bounds)
```

Comment:
Instead of having to input `parameter_bounds`, can't we just take the prior ranges as our parameter bounds by default?

Reply:
Yup, that would better. But how do I get ranges from the prior dict?

Reply:
I _think_ what we can do is read priors corresponding to certain parameters from the `priors` dictionary and read off it's min and maximum.

The following code snippet might help.

```python
In [4]: prior = bilby.gw.prior.UniformSourceFrame(minimum=10, maximum=100, name="luminosity_distance")

In [5]: prior
Out[5]: UniformSourceFrame(minimum=10.0, maximum=100.0, cosmology=FlatLambdaCDM(name="Planck15", H0=67.7 km / (Mpc s), Om0=0.307, Tcmb0=2.725 K, Neff=3.05, m_nu=[0.   0.   0.06] eV, Ob0=0.0486), name='luminosity_distance', latex_label='$d_L$', unit=Unit("Mpc"), boundary=None)

In [6]: prior.minimum
Out[6]: 10.0

In [7]: prior.maximum
Out[7]: 100.0

```

Reply:
This should work for most parameters, but it will fail (for sure) in those parameters which have a `DeltaFunction` prior.

Reply:
We can leave those parameters from maximization, right?

Reply:
This way, the user will just have to give the prior and the parameter bounds, as well as the parameters to be maximized over, are both decided by our program.

Reply:
Is that a good idea?

Reply:
Yes! I think it makes sense to start off with this. There might be some corner cases (there might be other priors which don't have a minimum or maximum), but we will get to that. How would you deal with constrained priors btw? Perhaps we can just think of that later.